addons/point_of_sale/static/src/js/Screens/ClientListScreen/ClientListScreen.js

```diff
@@ -191,7 +191,12 @@ odoo.define('point_of_sale.ClientListScreen', function(require) {
         async getNewClient() {
             var domain = [];
             if(this.state.query) {
-                domain = [["name", "ilike", this.state.query + "%"]];
+                domain = [
+                    '|','|',
+                    ["display_name", "ilike", this.state.query],
+                    ["ref", "=", this.state.query],
```

Comment:
I don't really see the relevance of searching based on the `ref`. It's not used and shown once in the front end

Reply:
Thanks for your feedback.
Yes, I agree. I removed it.
Thanks,